src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -255,6 +226,59 @@ public virtual PipeTransmissionMode ReadMode
         // ---- PAL layer ends here ----
         // -----------------------------
 
+        private static string s_pipeDirectoryPath;
+
+        private static string GetPipeDirectoryPath()
+        {
+            string path = s_pipeDirectoryPath;
+            if (path == null)
```

Comment:
Done.  I also changed it to recreate the directory each time if it didn't exist, rather than assuming that just because it existed when the static was set that it still exists.